File: Solutions/1694. Reformat Phone Number/1694. Reformat Phone Number.py
class Solution:
    def reformatNumber(self, number: str) -> str:
        tmp=""
        for _,c in enumerate(number):
            if c!=' ' and c!='-':
                tmp+=c
        res=""
        i=0
        while i<len(tmp):
            if len(tmp)-i==4:
                res+=tmp[i]
                res+=tmp[i+1]
                i+=2
            elif len(tmp)-i==2:
                res+=tmp[i]
                res+=tmp[i+1]
                break
            elif len(tmp)-i==3:
                res+=tmp[i]
                res+=tmp[i+1]
                res+=tmp[i+2]
                break
            else:
                res+=tmp[i]
                res+=tmp[i+1]
                res+=tmp[i+2]
                i+=3
            res+='-'
        return res

File: Solutions/1694. Reformat Phone Number/1694. Reformat Phone Number.go
func reformatNumber(number string) string {
    var tmp string
    for _,c:=range(number){
        if c!=' '&&c!='-'{
            tmp+=string(c)
        }
    }
    var res string
    for i:=0;i<len(tmp);{
        if(len(tmp)-i==4){
            res+=string(tmp[i])
            res+=string(tmp[i+1])
            i+=2
        }else if(len(tmp)-i==2){
            res+=string(tmp[i])
            res+=string(tmp[i+1])
            break
        }else if(len(tmp)-i==3){
            res+=string(tmp[i])
            res+=string(tmp[i+1])
            res+=string(tmp[i+2])
            break
        }else{
            res+=string(tmp[i])
            res+=string(tmp[i+1])
            res+=string(tmp[i+2])
            i+=3
        }
        res+=string('-')
    }
    return res
}